lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -3,6 +3,14 @@ def safe_added_columns_without_default_value
     @safe_added_columns_without_default_value ||= []
   end
 
+  # This variable is used to track nested lock acquisition.
+  # Each element is an array of table objects with their lock modes.
+  # The order of the array represents the current call stack,
+  # where the most recent method call is the last element.
+  def safely_acquire_lock_for_table_history
+    @safely_acquire_lock_for_table_history ||= []
+  end
```

Comment:
This seems way cleaner than using a thread variable. Not sure why I did that in the first place...